### DISCUSSION TITLE
VB to support consuming API that has required members.

### DISCUSSION THREAD

**Discussion by mrange** [dotnet/csharplang#6812] [Language Ideas] [answered]
Hi.

Perhaps this is already fixed but I wasn't able to google a solution and didn't find any issue matching my query. If this is a duplicate just close ASAP.

Anyway I have C# API that utilizes required members (from .NET7 IIRC). I can't get VB (running on .NET7) to consume the C# API successfully.

I am not concerned about being able to create APIs in VB that uses required members but rather consuming future C# APIs as this feature gets more and more used.

This is my example program. First the C# api
```csharp
namespace CsLib
{
  public class XX
  {
    public required int X { get; init; }
    public string? Y { get; set; }
  }
}
```

Then I try to consume this API in VB

```vb
Imports CsLib

Module Program
    Sub Main(args As String())
        ' Refuses to compile with: 
        '  error BC37319: 'Public Overloads Sub New()' requires compiler feature 'RequiredMembers', which is not supported by this version of the Visual Basic compiler.
        Dim x As New XX() With {.X = 1, .Y = "123"}
        Console.WriteLine("Hello World!")
    End Sub
End Module
```

The error message implies that there might be a newer version of the compiler that I could use but I wasn't able to find that the switch.  If there is but I missed my apologies for opening a duplicate ticket.

**Comment by HaloFour** [comment]
Looks like there's an issue to determine support on the Roslyn repo: https://github.com/dotnet/roslyn/issues/61435

**Comment by mrange** [comment, marked as answer]
Actually I thought I posted in roslyn but posted in C# lang for some reason which makes little sense. Thanks for finding the issue.

**Comment by mrange** [comment]
Creating it in vblang instead which makes more sense: https://github.com/dotnet/vblang/issues/619
